### PR TITLE
Move k8s config files to their respective component dir

### PR DESCRIPTION
**Pull request by ericbottard**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -16,6 +16,11 @@ dockerize:
 	$(MAKE) -C http-gateway			dockerize
 	$(MAKE) -C topic-controller		dockerize
 
+kubectl-apply:
+	$(MAKE) -C kubernetes-crds		kubectl-apply
+	$(MAKE) -C function-controller	kubectl-apply
```
Comment (glyn) [inline Makefile]: Indentation looks wrong, at least on github.

File: function-controller/Makefile [hunk 2]
```diff
@@ -35,3 +35,5 @@ debug-dockerize: $(GO_SOURCES) ../vendor
 ../glide.lock: ../glide.yaml
 	cd .. && glide up -v --force
 
+kubectl-apply:
+	kubectl apply -f config/function-controller-deployment.yaml
```
Comment (glyn) [inline function-controller/Makefile]: Missing newline according to github.

File: kubernetes-crds/Makefile [hunk 3]
```diff
@@ -52,3 +52,7 @@ hack/vendor: hack/glide.lock
 hack/glide.lock: hack/glide.yaml
 	# Note the absence of -v
 	cd hack && glide up
+
+kubectl-apply:
+	kubectl apply -f config/function-resource.yaml
+	kubectl apply -f config/topic-resource.yaml
```
Comment (glyn) [inline kubernetes-crds/Makefile]: Missing newline.

File: topic-controller/Makefile [hunk 4]
```diff
@@ -30,3 +30,5 @@ debug-dockerize: $(GO_SOURCES) ../vendor
 ../glide.lock: ../glide.yaml
 	cd .. && glide up -v --force
 
+kubectl-apply:
+	kubectl apply -f config/topic-controller-deployment.yaml
```
Comment (glyn) [inline topic-controller/Makefile]: Missing newline.

File: Makefile (hunk 1 above)
Comment (ericbottard) [reply Makefile]: Add `?ts=4` to the github URL and everything falls into place :)

I can change to spaces, but I know people on this project prefer tabs (I won't name names)

**Pull request merged**